### DISCUSSION TITLE
why does it render the page inside an iframe

### DISCUSSION THREAD

**Discussion by lili21** [niklasvh/html2canvas#2987] [General]
can't we just render it on the canvas?

**Comment by k1w1** [comment]
I have this question too. I notice that if I hack the code a little to avoid the iframe and generate the canvas directly from the element on the page it still works, and is about a second faster.

I assume there was some corner case that required the use of the iframe. It would be interesting to know what it was.

**Reply by freshteapot** [reply]
@k1w1 by any chance do you have your hack floating around in a branch somewhere?

Currently wondering how to manipulate the DocumentCloner to my will, but your comment gives me hope.

**Reply by k1w1** [reply]
Unfortunately I didn't save my code because I didn't end up using this approach. I didn't spend long on it, so whatever I did must have been pretty simple. I was interested in being able to render a lot of small things very quickly, so performance was the main consideration.
